nexus/src/app/background/tasks/read_only_region_replacement_start.rs

```diff
@@ -209,14 +210,16 @@ mod test {
             datastore.clone(),
         );
 
-        // Record which datasets map to which zpools for later
+        // Record which crucible datasets map to which zpools for later
 
         let mut dataset_to_zpool: BTreeMap<ZpoolUuid, DatasetUuid> =
             BTreeMap::default();
 
         for zpool in disk_test.zpools() {
             for dataset in &zpool.datasets {
```

Comment:
Take it or leave it: it might be useful to make `zpool.datasets` non-pub and have a method that iterates just the Crucible datasets.  That would ensure that we've found all possible uses of `datasets` that needed to be updated in this PR and also makes it easier in the future (e.g., if we add some other kind of dataset, we won't need to go change all these callers again).

Reply:
Done in https://github.com/oxidecomputer/omicron/pull/7982/commits/36d2d04675a7a6a43bbae732f65d7d0fc35a1309.

I built this originally as an iterator, before realizing: Actually, tests kinda *are* assuming that we'll have at most one dataset kind per zpool. 

So I'm exposing a helper function that provides that lookup of "find a dataset of a specific kind", validating that there is exactly one. If that assumption breaks, tests will break pretty clearly, and we can keep updating the `TestZpool` implementation.